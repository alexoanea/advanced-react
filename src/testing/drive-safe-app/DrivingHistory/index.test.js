import React from "react"
import { render, screen } from "@testing-library/react"
import DrivingHistory from "."

describe("DrivingHistory", ()=>{

    test("test confirmed trip", () =>{
        const trips = [{
            id: "trip-03",
            date: "5th July 2020",
            incidents: 35,
            distance: 8,
            confirmed: true,
          },]
        render(<DrivingHistory trips={trips}/>)
        const text = screen.queryByText('Confirmed')
        expect(text).toBeTruthy
    })
    test("test unconfirmed trip", () =>{
        const trips = [{
            id: "trip-03",
            date: "5th July 2020",
            incidents: 35,
            distance: 8,
          },]
        render(<DrivingHistory trips={trips}/>)
        const text = screen.queryByText('Confirmed')
        expect(text).toBeFalsy
        const button = screen.queryByRole('button')
        expect(button.textContent).toBe('Confirm Trip')
    })

    test("driving history snapshot test ", () =>{
      const trips = [
        {
            id: "trip-01",
            date: "1st July 2020",
            incidents: 5,
            distance: 100,
            confirmed: true,
        },
        {
            id: "trip-02",
            date: "3rd July 2020",
            incidents: 15,
            distance: 100,
            confirmed: true,
        },
        {
            id: "trip-03",
            date: "5th July 2020",
            incidents: 35,
            distance: 8,
        }
    ]
    const { baseElement } = render(<DrivingHistory trips={trips}/>)
      expect(baseElement).toMatchSnapshot()
  })
})