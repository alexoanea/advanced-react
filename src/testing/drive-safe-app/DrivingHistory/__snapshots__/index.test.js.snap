// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrivingHistory driving history snapshot test  1`] = `
<body>
  <div>
    <div
      style="max-width: 900px;"
    >
      <h2
        class="mb-4"
      >
        Your Trips
      </h2>
      <table
        class="table"
      >
        <thead>
          <tr>
            <th>
              #
            </th>
            <th>
              Date
            </th>
            <th>
              Distance (miles)
            </th>
            <th>
              Incidents
            </th>
            <th>
              Score
            </th>
            <th />
          </tr>
        </thead>
        <tbody>
          <tr
            data-testid="trip-row"
            style="background: rgb(20, 255, 20);"
          >
            <td>
              1
            </td>
            <td>
              1st July 2020
            </td>
            <td>
              100
            </td>
            <td>
              5
            </td>
            <td>
              95
            </td>
            <td>
              Confirmed
            </td>
          </tr>
          <tr
            data-testid="trip-row"
            style="background: rgb(46, 230, 241);"
          >
            <td>
              2
            </td>
            <td>
              3rd July 2020
            </td>
            <td>
              100
            </td>
            <td>
              15
            </td>
            <td>
              85
            </td>
            <td>
              Confirmed
            </td>
          </tr>
          <tr
            data-testid="trip-row"
            style="background: rgb(255, 71, 71);"
          >
            <td>
              3
            </td>
            <td>
              5th July 2020
            </td>
            <td>
              8
            </td>
            <td>
              35
            </td>
            <td>
              0
            </td>
            <td>
              <button
                class="btn btn-dark btn-sm"
                type="button"
              >
                Confirm Trip
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</body>
`;
